### PR TITLE
fix(tabby): fix swagger's local server use local port

### PR DESCRIPTION
**Pull request by hufeng**
```
 servers(
        (url = "https://playground.app.tabbyml.com", description = "Playground server"),
        (url = "http://localhost:8080", description = "Local server"),
    ),
```
The default is to use port 8080, but serve can customize the port through --port, resulting in request errors

### CODE REVIEW

File: crates/tabby/src/serve/mod.rs [hunk 1]
```diff
@@ -161,8 +160,20 @@ pub async fn main(config: &Config, args: &ServeArgs) {
     }
 
     info!("Starting server, this might takes a few minutes...");
+
+    // swagger add local server
+    let mut doc = ApiDoc::openapi();
```
Comment (wsxiaoys) [inline crates/tabby/src/serve/mod.rs:165]: extract this into a function like `add_localhost_server(doc: Doc) -> Doc`

**Review by wsxiaoys — CHANGES_REQUESTED** [review]
Thanks for the PR, one nit

File: crates/tabby/src/serve/mod.rs (hunk 1 above)
Comment (hufeng) [reply crates/tabby/src/serve/mod.rs:165]: 👌，Let me refactor the code ～

File: crates/tabby/src/serve/mod.rs [hunk 2]
```diff
@@ -161,8 +160,13 @@ pub async fn main(config: &Config, args: &ServeArgs) {
...
+    add_localhost_server(&mut doc, args.port);
```
Comment (wsxiaoys) [inline crates/tabby/src/serve/mod.rs:166]:
```suggestion
    let doc = add_localhost_server(APIDoc::openapi(), args.port);
```

Will something like this work?

Comment (hufeng) [reply crates/tabby/src/serve/mod.rs:166]: yes, it more clear

**Review by wsxiaoys — APPROVED** [review]

**Pull request merged**